src/awkward/contents/emptyarray.py

```diff
@@ -156,42 +156,40 @@ def _getitem_next(self, head, tail, advanced):
         elif head is Ellipsis:
             return self._getitem_next_ellipsis(tail, advanced)
 
-        elif isinstance(head, ak._v2.index.Index64):
+        elif isinstance(head, ak.index.Index64):
             if not head.nplike.known_shape or head.length == 0:
                 return self
             else:
-                raise ak._v2._util.indexerror(self, head.data, "array is empty")
+                raise ak._util.indexerror(self, head.data, "array is empty")
 
-        elif isinstance(head, ak._v2.contents.ListOffsetArray):
-            raise ak._v2._util.indexerror(self, head, "array is empty")
+        elif isinstance(head, ak.contents.ListOffsetArray):
+            raise ak._util.indexerror(self, head, "array is empty")
 
-        elif isinstance(head, ak._v2.contents.IndexedOptionArray):
-            raise ak._v2._util.indexerror(self, head, "array is empty")
+        elif isinstance(head, ak.contents.IndexedOptionArray):
+            raise ak._util.indexerror(self, head, "array is empty")
 
         else:
-            raise ak._v2._util.error(AssertionError(repr(head)))
+            raise ak._util.error(AssertionError(repr(head)))
 
     def num(self, axis, depth=0):
         posaxis = self.axis_wrap_if_negative(axis)
 
         if posaxis == depth:
             out = self.length
-            if ak._v2._util.isint(out):
+            if ak._util.isint(out):
                 return np.int64(out)
             else:
                 return out
         else:
-            out = ak._v2.index.Index64.empty(0, self._nplike)
-            return ak._v2.contents.numpyarray.NumpyArray(out, None, None, self._nplike)
+            out = ak.index.Index64.empty(0, self._nplike)
+            return ak.contents.numpyarray.NumpyArray(out, None, None, self._nplike)
```

Comment:
I guess, eventually we want to use `ak.contents.NumpyArray` instead of `ak.contents.numpyarray.NumpyArray`. The same applies to other `*Array`s

Reply:
Right, that would be a good thing to unify (later). I think we'll be able to take all of the short forms, e.g. `ak.contents.NumpyArray` instead of `ak.contents.numpyarray.NumpyArray`. The only reason I can think of for the long forms is if there's a cyclic reference and `awkward/contents/__init__.py` isn't fully evaluated yet, but that can only happen in module-level references.

I think there's something that I don't understand about Python's (or Python 3's) submodule imports because @henryiii said that it should be possible to make `awkward/__init__.py`'s imports order-independent. On another occasion, he said something about Vector being limited because Awkward wasn't using relative imports consistently. I think there's a rule I or we need to learn about that, but then it will be a mechanical or semi-mechanical process to apply it.

(This, by the way, might become its own issue; it doesn't affect PR #1721 dropping v1.)